lib/types/Vector2.lua

```diff
@@ -0,0 +1,58 @@
+local assign = import("../assign")
+
+local Vector2 = {}
+
+setmetatable(Vector2, {
+	__tostring = function()
+		return "Vector2"
+	end,
+})
+
+local prototype = {}
+
+local metatable = {}
+metatable.type = Vector2
+
+function metatable:__index(key)
+	local internal = getmetatable(self).internal
+
+	if internal[key] ~= nil then
+		return internal[key]
+	end
+
+	if prototype[key] ~= nil then
+		return prototype[key]
+	end
+
+	error(string.format("%s is not a valid member of Vector2", tostring(key)), 2)
+end
+
+function metatable:__add(other)
```

Comment:
It'd be nice to have subtraction. I'm not positive on the exact semantics of `__mul` and `__div`, but I'm pretty sure you can do scalar division and multiplication, which Roblox does implement.